torch/_inductor/lowering.py

```diff
@@ -2543,7 +2546,14 @@ def apply_constraint(idx, arg, fx_arg):
                     stride = ceildiv(stride, ALIGNMENT) * ALIGNMENT
 
                 out_strides[i] = stride
-            return ir.ExternKernel.require_exact_strides(arg, out_strides)
+
+            orig_size = arg.get_size()
+            for dim in expanded_dims:
+                arg = slice_(arg, dim, 0, 1)
+
+            out = ir.ExternKernel.require_exact_strides(arg, out_strides)
```

Comment:
cc @exclamaforte if you change:

`out = ir.ExternKernel.require_exact_strides(arg, out_strides)`

to 

```
arg = ir.ExternKernel.realize_input(arg)
out = ir.ExternKernel.require_exact_strides(arg, out_strides)
``` 

here you will get 

```
(Pdb) p node1.read_writes.writes
OrderedSet([MemoryDep('buf0', 1015104*d0 + 1008*d1 + d2, {d0: 32, d1: 1007, d2: 1008})])
(Pdb) p node2.read_writes.reads
OrderedSet([MemoryDep('buf0', 1015104*d0 + 1008*d1 + d2, {d0: 32, d1: 1007, d2: 1007})])
```

it's not a problem today in this lowering but just makes it tricky to get the exact right order of realizing/requiring stides/slicing/expanding. i had to rewrite a couple times